### ISSUE TITLE
[FEAT] Add detection of NPUs / AI accelerators and their capabilities.

### ISSUE THREAD

**Issue by ghost** [fastfetch-cli/fastfetch#1012]
## Wanted Features:

Detect NPUs and other AI accelerator devices and their features, for example: manufacturer name, device name, architecture, frequency, number of cores, and supported AI frameworks.

## Motivation:

With the growing popularity of AI, more manufacturers and more users have devices specifically for increasing AI performance, like dedicated AI accelerator PCIe cards and NPUs integrated into desktop processors and mobile SoCs. Soon, those devices may become almost as important for some users as the CPU or GPU.

Please note that I don't currently have such a device myself, so I won't be able to help in testing it, but I thought that fastfetch has some much less important modules, and this one is, in my opinion, missing.
If it is not possible for you to implement it at the moment, feel free to leave it for a future time.

**Comment by CarterLi** [comment]
> I don't currently have such a device myself

Me either.

**Comment by xiaoran007** [comment]
Maybe I can help add this feature. It would be easy to get this information from the pci bus.
But the main problem is that the current NPU drivers are not complete in their functionality (especially on linux), so more specific information such as the number of neural network cores, is not readily available at this time.

**Comment by Froglodyte** [comment]
Hey I've actually already implemented this for personal use on Linux. I never knew this was an open issue until now since its all the way at the bottom. I can clean up my code and make a PR. I'm dualbooting so I can also add Windows support later

**Comment by CarterLi** [comment]
> Hey I've actually already implemented this for personal use on Linux. I never knew this was an open issue until now since its all the way at the bottom. I can clean up my code and make a PR. I'm dualbooting so I can also add Windows support later

How did you do that? Enumerating PCI devices? If so, you can only get the device name but nothing else. If so, it's not interesting to me TBO.

**Comment by Froglodyte** [comment]
Yes for now its just that. What other information would you like to retrieve? I can check if its doable

**Comment by CarterLi** [comment]
@Froglodyte  https://www.intel.com/content/www/us/en/products/sku/245529/intel-core-ultra-9-processor-386h-18m-cache-up-to-4-90-ghz/specifications.html
